.github/workflows/docker.yaml

```
@@ -0,0 +1,67 @@
+name: Docker Build Action
+
+on:
+  pull_request:
+    branches:
+      - master
+  push:
+    branches:
+      - master
+  release:
+    types:
+      - published
+
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME: ${{ github.repository }}
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v4
+
+    - name: Figure out if running fork PR
+      id: fork
+      run: '["${{ secrets.DOCKER_REGISTRY_TOKEN }}" == ""] && echo "::set-output name=is_fork_pr::true" || echo "::set-output name=is_fork_pr::false"'
```

Let's use the "new" approach for dynamically setting variables across the steps:

```suggestion
      run: |
        ["${{ secrets.DOCKER_REGISTRY_TOKEN }}" == ""] && echo "is_fork_pr=true" >> $GITHUB_ENV || echo "is_fork_pr=false" >> $GITHUB_ENV
```